src/main/kotlin/net/theevilreaper/dartpoet/code/writer/ConstructorWriter.kt

```diff
@@ -5,42 +5,73 @@ import net.theevilreaper.dartpoet.code.CodeWriter
 import net.theevilreaper.dartpoet.code.emitParameters
 import net.theevilreaper.dartpoet.function.constructor.ConstructorSpec
 import net.theevilreaper.dartpoet.util.NEW_LINE
+import net.theevilreaper.dartpoet.util.SEMICOLON
 
 class ConstructorWriter {
 
-    fun emit(spec: ConstructorSpec, codeWriter: CodeWriter) {
-        for (modifier in spec.modifiers) {
-            codeWriter.emit("${modifier.identifier}·")
+    fun emit(spec: ConstructorSpec, writer: CodeWriter) {
+        if (spec.modifiers.contains(DartModifier.CONST)) {
+            writer.emit("${DartModifier.CONST.identifier}·")
         }
 
         if (spec.isFactory) {
-            codeWriter.emit("${DartModifier.FACTORY.identifier}·")
+            writer.emit("${DartModifier.FACTORY.identifier}·")
         }
 
-        codeWriter.emit(spec.name)
+        writer.emit(spec.name)
 
         if (spec.isNamed) {
-            codeWriter.emit(".${spec.named}")
-            spec.parameters.emitParameters(codeWriter) {
-                it.write(codeWriter)
+            writer.emit(".${spec.named}")
+        }
+
+        if (!spec.hasParameters) {
+            writer.emit("()$SEMICOLON")
```

Comment:
Maybe thats a thing to do for like that static stuff. Also complete constants

Reply:
Yes this could be a constant